technical-docs/designs/infra/rollback.md

```diff
@@ -0,0 +1,218 @@
+approval after the successful gate has been triggered.
+
+### 3.3 Terraform Backend Rollback
+
+With _Azure DevOps Pipeline Environments_, we can monitor Azure Web Apps, and
+Kubernetes clusters as resources of deployments. We can also track the health
+state of workloads and services within the cluster. Build deployments are
+associated with PR git hashes and we can use this to identify a previous healthy
+state to roll back our environment to.
+
+![](images/environmentService.png)
+
+Let's navigate back to our _Rollback Check_ stage and add a task to Download
+Pipeline Artifacts. Here we can download a previous build artifact that can be
+tagged to a specific version. The build version number can be passed in from a
+variable group that tracks successful builds of your resources.
+
+![](images/downloadartifact.png)
+
+Now you have the option to run pre-validation tests to assure the
+`terraform apply` will be successful. For simplicity we run `terraform init`,
+`plan` in this stage. When configuring the terraform tasks, use your variable
+group to pass the values for the `Resource group`, `storage account` and
+`Container`. The `key` will be the previous terraform state key produced in from
+the prior build artifact. Below is an example of a YAML:
+
+```
+steps:
+- task: ms-devlabs.custom-terraform-tasks.custom-terraform-release-task.TerraformTaskV1@0
+  displayName: 'Terraform : apply'
+  inputs:
+    backendServiceArm: '<Sub_ID>'
+    backendAzureRmResourceGroupName: 'nr-aks-hlf'
+    backendAzureRmStorageAccountName: nrhype
+    backendAzureRmContainerName: nrhype
+    backendAzureRmKey: $(Previous_tfstate_key_git_version)
+```
+
+#### Configure a Post-deployment Condition Approval
+
+Finally we configure a post-deployment approval that selects a user who can
+approve or reject deployments between our Rollback Check stage and Rollback
+Apply. This allows a human in the loop prior to our automation deploying to our
+environment.
+
+![](images/postdeploy.png)
+
+## 4. Dependencies
+
+This design for a Rollback Terraform Deployments will leverage the following:
+
+- [Bedrock Pre-Reqs: az cli | terraform | golang | fabrikate ](https://github.com/microsoft/bedrock/tree/master/tools/prereqs)
+- [Azure DevOps Environments](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/environments?view=azure-devops)
+- [Terraform](https://github.com/haishicorp/terraform)
+- [Release Gates](https://docs.microsoft.com/en-us/azure/devops/pipelines/release/approvals/gates?view=azure-devops)
+
+## 5. Risks & Mitigations
+
+**Risks & Limitations:**
+
+- Redeployment of stateful services in your environment are not accounted for in
+  this guide.
+- Complications arise when incompatible database changes during failed apply
+- Disaster recovery operability in respective to a fail production environment
+
+## 6. Documentation
+
+Yes, Documentation will need to be added to the guide docs fo.r rollback
```

Comment:
@NathanielRose just seeing that your intention is to put this in `guides\*`. 👍🏾

Spelling issue here